tests/IntegrationTests/WebAppUiTests/AutomatedUiTests.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Identity.Web.Test.Common;
+using Microsoft.Identity.Web.Test.LabInfrastructure;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using Xunit;
+
+namespace WebAppUiTests
+{
+#if !FROM_GITHUB_ACTION
+    public class AutomatedUiTests
+    {
+        [Fact]
+        public async Task ChallengeUser_SignInSucceedsTestAsync()
+        {
+            // Arrange
+            LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+
+            ChromeOptions options = new ChromeOptions();
+            // ~2x faster, no visual rendering
+            // comment-out below when debugging to see the UI automation
+            options.AddArguments(TestConstants.Headless);
+            IWebDriver driver = new ChromeDriver(options);                  
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);            
+
+            // Act
+            Trace.WriteLine("Starting Selenium automation: web app sign-in & call Graph");
+            driver.Navigate()
+               .GoToUrl("https://webapptestmsidweb.azurewebsites.net/MicrosoftIdentity/Account/signin");
+            PerformLogin(driver, labResponse.User);
+
+            // Assert
+            Assert.Contains(labResponse.User.Upn, driver.PageSource);
+            Assert.Contains(TestConstants.PhotoLabel, driver.PageSource);
```

Comment:
other ideas for asserts are welcome, but if we signed in correctly & called graph, we get the upn appearing and the photo label for graph.